control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -212,6 +212,11 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 	r.Log.Info("found release info for image", "releaseImage", hostedControlPlane.Spec.ReleaseImage, "info", releaseImage, "componentImages", releaseImage.ComponentImages(), "componentVersions", componentVersions)
 
+	if hostedControlPlane.Status.Version == "" {
+		hostedControlPlane.Status.Version = releaseImage.Version()
+		return ctrl.Result{}, patchHelper.Patch(ctx, hostedControlPlane)
```

Comment:
What's the reason to return early rather than proceed w/ the updated structure?

Reply:
This is the pattern I've followed with controller-runtime since as far as I can remember. When reconciling a resource, if you persist a change to the resource, return and wait for the cache to pick up the updated resource and trigger a new sync. This way if your update causes other changes (say via mutating webhook, etc) you get the most up to date resource from the cache when syncing again.